### PR TITLE
ARROW-16358: [CI][Dev] Allow archery crossbow to generate a CSV report for nightly builds

### PR DESCRIPTION
**Pull request by raulcd**
This PR will allow us to generate a CSV with the nightly builds status. This will be used in order to generate a static page with the different builds information and will allow us to analyze historical information from the different builds like since when they have been failing, last successful build commit, some tags, etcetera.

An example of a CSV generated can be seen on the following gist:
https://gist.github.com/raulcd/1799c8499ef228dfdf1ab0d986367d2a

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
> * I am not storing the date as part of the CSV because the filename generated will be something like: `nightly-packaging-2022-04-25-0.csv `.

I think that is fine. But I am wondering, it might be useful to keep track of the exact commit that is being tested in the build? For example in case you want to check if a certain fix was already included in the run or not (currently only the date is not that informative, given a date is still a broad range and depending on when it is exactly scheduled on that day)

File: dev/archery/archery/crossbow/reports.py [hunk 1]
```diff
@@ -83,6 +84,29 @@ def tasks(self):
...
+                task_status.combined_state,
+                task_status.build_links,
+                self.branch_url(task.branch),
+                task.ci,
+                # We want this to be serialized as a dict instead
+                # of an orderedict.
+                {k: v for k, v in task.params.items()},
+                task.template,
+                # Arrow repository commit
+                self.job.target.head
```
Comment (jorisvandenbossche) [inline dev/archery/archery/crossbow/reports.py:106]: I asked about a commit, but I see here that it is already included! It seems that I missed that there was more content in https://gist.github.com/raulcd/1799c8499ef228dfdf1ab0d986367d2a (only the first three columns are visible ;))

**Comment by jorisvandenbossche** [comment]
> I am not generating a header row for the CSV.

It wouldn't be that hard to do, I think, and could improve usability a bit given that there are quite some columns in there? (eg another attribute `rows_header` that returns a single list that can be written first to the csv file)

File: dev/archery/archery/crossbow/reports.py (hunk 1 above)
Comment (raulcd) [reply dev/archery/archery/crossbow/reports.py:106]: yes, the view on the gist for CSV is not great :)

**Comment by raulcd** [comment]
> It wouldn't be that hard to do, I think, and could improve usability a bit given that there are quite some columns in there? (eg another attribute `rows_header` that returns a single list that can be written first to the csv file)

I have added the CSV header and have updated the gist with the updated outcome of running the command locally:
```
$ archery crossbow -t $GITHUB_TOKEN report-csv --no-fetch --save nightly-packaging-2022-04-25-0

```

File: dev/archery/archery/crossbow/reports.py [hunk 2]
```diff
@@ -30,6 +30,17 @@
 # TODO(kszucs): use archery.report.JinjaReport instead
 class Report:
 
+    ROW_HEADERS = [
+        "Task name",
```
Comment (jorisvandenbossche) [inline dev/archery/archery/crossbow/reports.py:34]: We could also use names like `"task_name"` instead (lowercase, no spaces), which are generally easier to use when afterwards playing with the data with pandas or R or something alike

Comment (raulcd) [reply dev/archery/archery/crossbow/reports.py]: done! Thanks!

**Review by jorisvandenbossche — APPROVED** [review]
Thanks!

**Pull request closed without merging**